src/hotspot/cpu/aarch64/c1_LIRGenerator_aarch64.cpp

```diff
@@ -277,18 +277,20 @@ void LIRGenerator::cmp_reg_mem(LIR_Condition condition, LIR_Opr reg, LIR_Opr bas
 
 
 bool LIRGenerator::strength_reduce_multiply(LIR_Opr left, jint c, LIR_Opr result, LIR_Opr tmp) {
-
-  if (is_power_of_2(c - 1)) {
-    __ shift_left(left, exact_log2(c - 1), tmp);
+  unsigned int u_value = (juint)c;
```

Comment:
```suggestion
  juint u_value = (juint)c;
```

Reply:
I don't have hardware for arm32 :(

Reply:
We can ask @bulasevich (also see https://wiki.openjdk.org/display/HotSpot/Ports).

Reply:
Should I revert arm32 changes ? Maybe it can be done with another JBS issue.

Reply:
@shipilev can you do build+test on arm32, please?

Reply:
Sorry for delay. I will check arm32.

Reply:
All right. The ARM32 build is fine and jtreg testing shows no regressions. Thanks for fixing this for all ports!

Reply:
I need one more approval to integrate this :)